app.py

```diff
@@ -19,17 +20,28 @@ def home() -> str:
 
 @app.route('/animals', methods=['GET'])
 def index() -> Response:
-    animals = Animal.query.all()
-    return jsonify({"animals": [AnimalResponse.model_validate(animal).model_dump(mode='json') for animal in animals]})
+    name = request.args.get('name')
+    animals = Animal.query.filter_by(name=name).all() if name else Animal.query.all()
+    response = {'animals': []}
 
+    for animal in animals:
+        model = AnimalResponse.model_validate(animal)
+        age = model.calculate_age()
+        data = model.model_dump(mode='json')
+        data['age'] = age
+        response['animals'].append(data)
```

Comment:
You can simplify your code, if you'll add `age` to pydantic model as computed field
```
https://docs.pydantic.dev/2.0/usage/computed_fields/
```
And model_dump() will return response already with this field.